### PR TITLE
Add debugging for Dart VM timeout flake

### PR DESCRIPTION
**Pull request by vashworth**
Check what is available in the device's iOS DeviceSupport folder to check if symbols were properly fetched. Also, add some logging to track what status the debugger is in.

Debugging for https://github.com/flutter/flutter/issues/121231.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by vashworth** [comment]
I described this in the comments in the code, but FYI:

The expected files to be found in the device's iOS DeviceSupport include Symbols (directory), Info.plist, and .finalized.

If any of the expected files are missing or there are additional files (such as .copying_lock or .processing_lock), this may indicate the symbols may still be fetching or something went wrong when fetching them.

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart [hunk 1]
```diff
@@ -306,6 +309,8 @@ class IOSDeployDebugger {
   // (lldb) Process 6152 resuming
   static final RegExp _lldbProcessResuming = RegExp(r'Process \d+ resuming');
 
+  static final RegExp _symbolsPathPattern = RegExp(r'.*Symbol Path: ');
```
Comment (hellohuanlin) [inline packages/flutter_tools/lib/src/ios/ios_deploy.dart:312]: Can you add an example path (like the lines above)

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart [hunk 2]
```diff
@@ -376,6 +393,7 @@ class IOSDeployDebugger {
           _logger.printTrace(line);
           final bool attachSuccess = line == 'success';
           _debuggerState = attachSuccess ? _IOSDeployDebuggerState.attached : _IOSDeployDebuggerState.detached;
+          _logger.printTrace('Debugger state set to ${attachSuccess ? 'attached' : 'detached'}.');
```
Comment (hellohuanlin) [inline packages/flutter_tools/lib/src/ios/ios_deploy.dart:396]: Are these logs also print out to real developers? do we plan to remove these new logs after we fix this flake? If so may wanna add a `TODO` to remove them.

**Review by hellohuanlin — APPROVED** [review]

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart [hunk 3]
```diff
@@ -362,12 +368,25 @@ class IOSDeployDebugger {
           return;
         }
 
+        // Symbol Path: /Users/swarming/Library/Developer/Xcode/iOS DeviceSupport/16.2 (20C65) arm64e/Symbols
+        if (_symbolsPathPattern.hasMatch(line)) {
+          _logger.printTrace(line);
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/ios/ios_deploy.dart:373]:
```suggestion
          _logger.printTrace('Detected path to iOS debug symbols: "$line"');
```

**Review by christopherfujino — APPROVED** [review]
nit about adding a bit more context to one trace, but LGTM

**Pull request merged**